torch/csrc/inductor/aoti_torch/shim_common.cpp

```diff
@@ -0,0 +1,181 @@
+#include <c10/core/DeviceType.h>
+#include <c10/core/ScalarType.h>
+#include <c10/util/Exception.h>
+#include <torch/csrc/inductor/aoti_torch/c/shim.h>
+#include <torch/csrc/inductor/aoti_torch/utils.h>
+#include <torch/csrc/inductor/inductor_ops.h>
+#include <cstdint>
+#include <cstdio>
+#include <iostream>
+#include <memory>
+
+#ifndef AT_PER_OPERATOR_HEADERS
+#include <ATen/Functions.h>
+#else
+
+#include <ATen/ops/_addmm_activation.h>
+#include <ATen/ops/addmm.h>
+#include <ATen/ops/as_strided.h>
+#include <ATen/ops/bmm.h>
+#include <ATen/ops/convolution.h>
+#include <ATen/ops/empty_strided.h>
+#include <ATen/ops/from_blob.h>
+#include <ATen/ops/mm.h>
+
+#endif
+
+int32_t aoti_torch_device_type_cpu() {
+  return (int32_t)c10::DeviceType::CPU;
+}
+
+int32_t aoti_torch_device_type_cuda() {
+  return (int32_t)c10::DeviceType::CUDA;
+}
+
+int32_t aoti_torch_dtype_bfloat16() {
+  return (int32_t)c10::ScalarType::BFloat16;
+}
+
+int32_t aoti_torch_dtype_float16() {
+  return (int32_t)c10::ScalarType::Half;
+}
+
+int32_t aoti_torch_dtype_float32() {
+  return (int32_t)c10::ScalarType::Float;
+}
+
+int32_t aoti_torch_dtype_float64() {
+  return (int32_t)c10::ScalarType::Double;
+}
+
+int32_t aoti_torch_dtype_uint8() {
+  return (int32_t)c10::ScalarType::Byte;
+}
+
+int32_t aoti_torch_dtype_int8() {
+  return (int32_t)c10::ScalarType::Char;
+}
+
+int32_t aoti_torch_dtype_int16() {
+  return (int32_t)c10::ScalarType::Short;
+}
+
+int32_t aoti_torch_dtype_int32() {
+  return (int32_t)c10::ScalarType::Int;
+}
+
+int32_t aoti_torch_dtype_int64() {
+  return (int32_t)c10::ScalarType::Long;
+}
+
+AOTITorchError aoti_torch_delete_tensor_object(AtenTensorHandle tensor) {
+  AOTI_TORCH_CONVERT_EXCEPTION_TO_ERROR_CODE({
+    at::Tensor* t = reinterpret_cast<at::Tensor*>(tensor);
+    delete t;
```

Comment:
Ahh I love the smell of tiny heap allocation in the morning (@desertfire tells me that this isn't really visible on the internal bench, but this is probably one of the first things I'd try to dump in microoptimizing. cc @swolchok)